### ISSUE TITLE
I have problems to run osrm-backend with planet-latest

### ISSUE THREAD

**Issue by wendibus** [Project-OSRM/osrm-backend#1460]
Hi,

I have started osrm-extract for planet-latest on a i7 Ubuntu System with 24 GByte of RAM. 
(Ubuntu runs in a virtual machine of my Mac and uses 7 cores and 24 of 32 GByte RAM)

I use stxxl with a 10 GByte file on an ssd drive.

[info] Input file: map.osm
[info] Profile: profile.lua
[info] Threads: 7
[info] Using script profile.lua
[STXXL-MSG] STXXL v1.3.1 (release)
[STXXL-MSG] 1 disks are allocated, total space: 10000 MiB

df -h says
osrm@osrm-server:~/osrm$ df -h
Dateisystem    Größe Benutzt Verf. Verw% Eingehängt auf
/dev/sda1       768G    5,4G  724G    1% /
none            4,0K       0  4,0K    0% /sys/fs/cgroup
udev             13G    4,0K   13G    1% /dev
tmpfs           2,5G    1,2M  2,5G    1% /run
none            5,0M       0  5,0M    0% /run/lock
none             13G    156K   13G    1% /run/shm
none            100M    104K  100M    1% /run/user
/dev/sdc        2,0T    877G  1,1T   46% /media/osrm/Work
/dev/sdb1        11G     28M   11G    1% /media/osrm/Cache

top says

top - 15:25:38 up 8 days,  7:26,  5 users,  load average: 0,17, 0,28, 0,23
Aufgaben: 305 total,   2 running, 303 sleeping,   0 stopped,   0 zombie
%Cpu(s):  1,4 be,  0,2 sy,  0,0 ni, 98,3 un,  0,0 wa,  0,0 hi,  0,0 si,  0,0 st
KiB Mem:  25687764 total, 18076204 used,  7611560 free,   151880 buffers
KiB Swap: 26178556 total,    71764 used, 26106792 free. 13445740 cached Mem

  PID BENUTZER  PR  NI    VIRT    RES    SHR S  %CPU %MEM     TIME+ COMMAND      
 2412 osrm      20   0 1767092 271600  26668 S  13,3  1,1   1643:28 compiz       
 1195 root      20   0  356208  66436  10328 S   2,0  0,3  56:43.40 Xorg         
 3134 osrm      20   0  663516  24964  13872 S   0,7  0,1   6:38.38 gnome-termi+ 
 4625 osrm      20   0 1621516 1,276g   4252 S   0,7  5,2  81:33.46 osrm-extract 

but even after 81 hours there is no new file nothing nada. So I assume that there is no sense in running it any longer ... is there any way to see whether the program is running right or not?
It does work with smaller files

Thanks for any advice. 

PS: I entered 
#Für OSRM eingefügt:
vm.swappiness=0

in the file /etc/sysctl.conf

and cmake stated:
-- Building on a 64 bit system
-- Configuring OSRM in release mode
-- LTO working
-- Boost version: 1.54.0
-- Found the following Boost libraries:
--   date_time
--   filesystem
--   iostreams
--   program_options
--   regex
--   system
--   thread
--   unit_test_framework
-- Found Intel TBB
-- Looking for Luabind...
-- Found Luabind: /usr/lib/libluabind.so
-- Looking for LuaJIT 5.2
-- Could NOT find LUAJIT (missing:  LUAJIT_LIBRARIES LUAJIT_INCLUDE_DIR) 
-- Performing Test LUABIND_WORKS
-- Performing Test LUABIND_WORKS - Success
-- Luabind/Lua5.2 combination working with /usr/lib/x86_64-linux-gnu/liblua5.2.so
-- Looking for STXXL...
-- Found STXXL: /usr/lib/libstxxl.so
-- OpenMP support found. Linking just in case for stxxl
-- Looking for OSMPBF...
-- Found OSMPBF: /usr/lib/libosmpbf.a
-- Configuring done
-- Generating done
-- Build files have been written to: /home/osrm/osrm/osrm-backend/build

**Comment by wendibus** [comment]
I have tried something different now. Instead of running planet-latest.osm I have tried 

osrmn-extract with planet-latest.osm.pbf

Now I do receive an error:

osrm-extract planet-latest.osm.pbf 
[info] Input file: planet-latest.osm.pbf
[info] Profile: profile.lua
[info] Threads: 7
[info] Using script profile.lua
[STXXL-MSG] STXXL v1.3.1 (release)
[STXXL-MSG] 1 disks are allocated, total space: 200000 MiB
[info] Parsing in progress..
[info] input file generated by planet-dump-ng 1.1.0
[info] timestamp: 2015-03-02T02:00:01Z
[info] Using turn restrictions
[info] Found 3 exceptions to turn restrictions:
[info]   motorcar
[info]   motor_vehicle
[info]   vehicle
terminate called after throwing an instance of 'osmium::pbf_error'
  what():  PBF error: invalid BlobHeader size (> max_blob_header_size)
Abgebrochen (Speicherabzug geschrieben)

What should I try next? stxxl is 200 GByte sized

**Comment by springmeyer** [comment]
I'm curious why you are running within a VM? Did you consider running directly on your Mac hardware?

**Comment by wendibus** [comment]
The only reason is that I failed to install it with Mac OS X and I do need my Mac. I was aware that it would calculate for many hours. Thats the whole reason. If I could have installed it as easy with Mac OS X as with Linux I would have done so ...

Or maybe it is easy and I just failed as some others did :-((

**Comment by springmeyer** [comment]
Okay, thanks for the details. I run on OS X and have been starting to contribute to OSRM by working to make it easier to build. So I can help you get running. The caveat however is that I'm not aware personally of anyone processing the entire planet on OS X. That may just be because people rarely have macs large enough however. Can you create a new ticket specifically about the problem you hit with Mac builds? Put errors in gist.github.com entries and then link them to the ticket. this will make it easier to read and help make sense of the problems.

**Comment by wendibus** [comment]
Okay I will try to do so ... right now I have managed to build europe-latest with my virtual machine ...

I ran into a problem with cmake but I will open a ticket as proposed by you.

**Comment by TheMarex** [comment]
For the car profile I would recommend something around 120GB of RAM for processing the entire planet. You can find some recent numbers [here](https://github.com/Project-OSRM/osrm-backend/wiki/Data-processing-timings). Also that osmium error suggests that your PBF file might be corrupted.

**Comment by wendibus** [comment]
Thank you for your advice. I will try another planet-latest (mine was from March 2015) just to be on the safe side ...

Do I really need 120 GByte of RAM or could I use instead a huge .stxxl file and wait much longer? I thought that you need this stxxl as an alternative to RAM (if you do not have enough RAM but time ) ...

**Comment by TheMarex** [comment]
You need the stxxl and enough swap space configured. See https://github.com/Project-OSRM/osrm-backend/wiki/Building-and-Running-OSRM-on-EC2 on how to setup swap space.

**Comment by TheMarex** [comment]
@wendibus does it work now?

**Comment by wendibus** [comment]
Thanks for asking ... I tried it again on Wednesday ... I guess it is running now for approx 39 hours ...
It is swapping all the time ... Memory always close to 9X% but proceeding.

Right now I have reached:
Removing graph geometry while preserving topology
. 10% . 20% . 30% . 40% . 50% .

I do have the impression that the progress is not linear. That it takes longer and longer to reach the additional 10%. But I'm not in a hurry. The weather is beautiful and I will spend my time outdoors while the computer is running.
I'll writer whether it will be successfull or not when its ended one way or another ;-))

**Comment by wendibus** [comment]
Its Monday and we are talking about 60% . 70% . 
:-(

**Comment by emiltin** [comment]
i think that's going to take a very long time to complete..

**Comment by TheMarex** [comment]
@wendibus yeah and that isn't even the part of the code that takes a lot of time. On the 16 core machine I use, processing takes 6 hour for the car profile. Do you use your own profile?

**Comment by wendibus** [comment]
Hi,

I'm running the test with the standard car profile.
Today I reached 90% . 

But for the final version I will take a different profile. Right now I just want to have an impression how long it takes, to do something for the planet. When I was young we calculated a mandelbaum picture on the Commodre 64 and it took days and once the machine crashed but finally we succeeded. This here feels quite familiar ;-))
